### DISCUSSION TITLE
Configuration to add custom git commit trailers

### DISCUSSION THREAD

**Discussion by esamson** [scala-steward-org/scala-steward#3130] [Ideas]
Right now we use [`commits.message`](https://github.com/scala-steward-org/scala-steward/blob/v0.25.0/docs/repo-specific-configuration.md?plain=1#L134) to have commits generated by scala-steward play well with other git integrations we have in our CI pipeline. This mostly works except for commits generated by post-update hooks since we cannot control the commit message there.

It would be good to be able to configure the commit message for an update hook, like the **scalafmt** one for example. But maybe a more general approach is to have a configuration for a list of trailers that will be added to all git commit messages -- both the regular one and those from extra hooks.

I don't really mind the existing messages but I would like to be able to add specific content to be able to drive my other git integrations.

**Comment by fthomas** [comment]
I don't mind having repo configuration for adding commit trailers to [`CommitMsg`](https://github.com/scala-steward-org/scala-steward/blob/ea94b76ed633b10bc41936b3e9f0f2308ddaa3db/modules/core/src/main/scala/org/scalasteward/core/git/CommitMsg.scala).
